functions/bigtable/main_async.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START bigtable_functions_quickstart_asyncio]
+import asyncio
+
+import functions_framework
+from google.cloud.bigtable.data import BigtableDataClientAsync, ReadRowsQuery, RowRange
+
+event_loop = asyncio.new_event_loop()
+asyncio.set_event_loop(event_loop)
+
+
+# Setup: create a shared client within the context of the event loop
+async def create_client():
+    # create client in the asyncio event loop context to
+    # give background tasks a chance to initialize
+    return BigtableDataClientAsync()
+
+
+client = event_loop.run_until_complete(create_client())
```

Comment:
Not super familiar with async/await in Python, but would it be possible to use await instead of run_until_complete somehow? Maybe we can await for the client within the async function definition for the sample?

Reply:
Moving client creation inside the async function would mean setting up a new client on each invocation. If we want to share the same client across calls (which we should encourage), I believe we have to do something like this

Reply:
+1 to daniel